library/src/main/kotlin/me/thanel/swipeactionview/SwipeActionView.kt

```diff
@@ -852,8 +924,15 @@ class SwipeActionView : FrameLayout {
         } else {
             leftSwipeRipple.restart()
         }
+        if(isHalfwaySwipe) {
+            if (swipedRight) {
+                swipeGestureListener?.onSwipedHalfwayRight(this)
+            } else {
+                swipeGestureListener?.onSwipedHalfwayLeft(this)
+            }
+        }
 
-        val targetTranslationX = if (swipedRight) maxRightSwipeDistance else -maxLeftSwipeDistance
+        var targetTranslationX = if (swipedRight) maxRightSwipeDistance else -maxLeftSwipeDistance
```

Comment:
Add and use a function here which will calculate target position based on whether its halfway swipe by accessing first child width and using that.